ttkwidgets/tickscale.py

```diff
@@ -0,0 +1,633 @@
+            to = self._get_precision(end)
+            interv = self._get_precision(self._tickinterval)
+            resol = self._get_precision(self._resolution)
+            digits = max(self._digits, interv, resol, from_, to)
+            if digits != self._digits:
+                self._digits = digits
+                self._formatter = '{:.' + str(self._digits) + 'f}'
+                reinit = True
+        if 'variable' in kw:
+            self._var = kw['variable']
+            if not self._var:
+                self._var = tk.DoubleVar(self, self.get())
+                kw['variable'] = self._var
+            try:
+                self._var.trace_add('write', self._increment)
+            except Exception:
+                # backward compatibility
+                self._var.trace('w', self._increment)
+
+        self.scale.configure(**kw)
+
+        if 'from_' in kw or 'from' in kw or 'to' in kw:
+            self._extent = self.scale.cget('to') - self.scale.cget('from')
+            self._start = self.scale.cget('from')
+            reinit = True
+        if 'style' in kw:
+            self._style_name = kw['style']
+            if not self._style_name:
+                self._style_name = '%s.TScale' % (str(self.scale.cget('orient')).capitalize())
+            self._style_change()
+        if 'orient' in kw:
+            if kw['orient'] == 'vertical':
+                self._style_name = self._style_name.replace('Horizontal', 'Vertical')
+                if 'tickpos' not in kw:
+                    self._tickpos = 'w'
+            else:
+                self._style_name = self._style_name.replace('Vertical', 'Horizontal')
+                if 'tickpos' not in kw:
+                    self._tickpos = 's'
+            self.scale.configure(style=self._style_name)
+            reinit = True
+        if reinit:
+            self._init()
+
+    def config(self, cnf={}, **kw):
+        self.configure(cnf=cnf, **kw)
+
+    def get(self):
+        if self._digits >= 0:
+            return round(self.scale.get(), self._digits)
+        else:
+            return self.scale.get()
+
+    def convert_to_pixels(self, value):
+        """Convert value in the scale's unit into a position in pixels."""
+        percent = ((value - self._start) / self._extent)
+        return percent * (self.get_scale_length() - self._sliderlength) + self._sliderlength / 2
+
+    def _update_slider_length_horizontal(self):
+        """Measure the length of the slider and update the value of self._sliderlength."""
+        if not self.scale.identify(2, 2):
+            # to soon, wait for the scale to be properly displayed
+            # binding to <Map> event does not work, it can still be to soon to
+            # get any result from identify
+            self.after(10, self._update_slider_length_horizontal)
+        else:
+            w = self.scale.winfo_width()
+            i = 0
+            while i < w and 'slider' not in self.scale.identify(i, 2):
```

Comment:
Could you perhaps add some line comments here to explain what's happening? I *think* the code is going over the pixels where the slider reports to be, and then determines the actual length in pixels by going over each pixel and calling Tkinter to determine if the scale is actually in that pixel coordinate, but I'm not completely sure.

Reply:
Done. I am indeed using `self.scale.identify(x, y)` that returns the element (slider, trough, ...) to which the pixel (x, y) belongs to determine the length of the slider in pixels.